prql-compiler/src/parser/chumsky.rs

```diff
@@ -78,3 +89,37 @@ pub fn ident() -> impl Parser<char, Ident, Error = Simple<char>> {
         .chain(just('.').ignore_then(ident_part().or(star)).repeated())
         .map(Ident::from_path::<String>)
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    use insta::assert_yaml_snapshot;
+    #[test]
+    fn test_date() {
+        let date = just('@')
+            .ignore_then(digits(4))
```

Comment:
To be clear, `digits(4)` parses any number of base 4 digits (0/1/2/3), not 4 digits. You'll probably want to use `filter(|c| c.is_digit()).repeated().exactly(4)` instead.

Reply:
(this refers to https://github.com/PRQL/prql/pull/1652/files#diff-bb3391b38da46712662bacac63ebceb0265e1070a736620862193eed84fd1640R71-R76 — I think maybe this conflicts with a `chumsky` function name :) )

Reply:
Ah, ignore me then :grinning: